include/onnxruntime/core/common/inlined_containers_fwd.h

```diff
@@ -98,7 +99,7 @@ struct CalculateInlinedVectorDefaultInlinedElements {
   // happens on a 32-bit host and then fails due to sizeof(T) *increasing* on a
   // 64-bit host, is expected to be very rare.
   static_assert(
-      sizeof(absl::InlinedVector<T, 1>) <= kPreferredInlinedVectorSizeof,
+      sizeof(T) <= 256,
```

Comment:
this was inconsistent with the comment above. the updated check was copied from the LLVM source.
https://github.com/microsoft/onnxruntime/blob/3064fa7611b55154155a60b1af8bb596c8381780/include/onnxruntime/core/common/inlined_containers_fwd.h#L87-L91

@yuslepukhin FYI. let me know if there is some reason for original deviation

Reply:
You will need to make it consistent.:)

Reply:
So what is the motivation behind this specific change in inlined containers? Is there any metric besides LLVM does so?

Also why 256 instead of a named constant?

Another question why 256 is different from the kPreferredInlinedVectorSizeof?

The static_assert would no longer fire and we would get something bad on line 110.
If there is a contradiction between the comment and the code, the comment should prevail since it describes a valid use case. IMHO.

Reply:
> So what is the motivation behind this specific change in inlined containers? Is there any metric besides LLVM does so?
> Another question why 256 is different from the kPreferredInlinedVectorSizeof?

I think the comment explains the motivation, quoted here:
```
 // We want this assertion to trigger in pathological cases, but otherwise 
 // not be too easy to hit. To accomplish that, the cutoff is actually somewhat 
 // larger than kPreferredInlinedVectorSizeof (otherwise, 
 // `InlinedVector<InlinedVector<T>>` would be one easy way to trip it, and that 
 // pattern seems useful in practice). 
```

to me, it explains why the assertion was that way in the original version.

> Also why 256 instead of a named constant?

made it a constant.

Reply:
also, my original motivation for making this change was that I was trying to use an InlinedVector of a struct and tripping the static_assert